File: members/U02BQ6G1SQJ/programmers/problem/k진수에서 소수 개수 구하기.py
import math;

def convertNotation(srcNum : int, dstNotation : int) -> str:
    dstStr = "";

    while (srcNum > 0):
        (srcNum, dstDigit) = divmod(srcNum, dstNotation);
        dstStr = str(dstDigit) + dstStr;

    return dstStr;


def checkPrime(num: int) -> int:
    if (num == 1):
        return -1;

    """
    # for divisor in range(2, math.ceil(math.sqrt(num))):
    for divisor in range(2, num):
        if (num % divisor == 0):
            return -1;
    """

    divisor = 2;

    while (divisor ** 2 <= num):
        if (num % divisor == 0):
            return -1;

        divisor += 1;

    return num;

def solution(n : int, k : int) -> int:
    answer = 0;
    primeSet = set();
    pList = convertNotation(n, k).split("0");

    # print(pList);
    # print(convertNotation(1000000, 2));
    # print(len(convertNotation(1000000, 2)));

    for curP in pList:
        # print(checkPrime(int(curP)));

        if (not curP):
            continue;

        if (curP in primeSet):
            answer += 1;
            continue;

        primeResult = checkPrime(int(curP));

        if (primeResult != -1):
            primeSet.add(primeResult);
            answer += 1;

    # print(primeSet);

    return answer;

# print(solution(437674, 3));  # 3
# print(solution(110011, 10)); # 2
# print(solution(1, 10)); # 0
# print(solution(1, 2)); # 0

File: members/U02BWCHJA6Q/Programers/kakao_2022/k진수에서_소수_개수_구하기.js
function solution(n, k) {
  const isPrime = (n) => {
    if (n === 1) {
      return false;
    }

    if (n === 2) {
      return true;
    }

    for (let i = 2; i <= Math.sqrt(n); i++) {
      if (n % i === 0) {
        return false;
      }
    }

    return true;
  };

  return n
    .toString(k)
    .split('0')
    .filter((s) => {
      if (!s) {
        return false;
      }

      return isPrime(Number(s));
    }).length;
}
